consumer/consumererror/scrapeerrors.go

```diff
@@ -0,0 +1,64 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror
+
+import (
+	"fmt"
+)
+
+// ScrapeErrors contains multiple PartialScrapeErrors and can also contain generic errors.
+type ScrapeErrors struct {
+	errs              []error
+	failedScrapeCount int
+}
+
+// Add adds a PartialScrapeError with the provided failed count and error.
+func (s *ScrapeErrors) Add(failed int, err error) {
```

Comment:
I would call this `AddPartial`, and rename `AddRegular` to just `Add`?

Reply:
Done.